src/main/java/com/algorand/algosdk/v2/client/model/TransactionKeyreg.java

```diff
@@ -0,0 +1,62 @@
+package com.algorand.algosdk.v2.client.model;
+
+import java.util.Objects;
+
+import com.algorand.algosdk.v2.client.common.PathResponse;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Fields for a keyreg transaction. 
+ * Definition: 
+ * data/transactions/keyreg.go : KeyregTxnFields 
+ */
+public class TransactionKeyreg extends PathResponse {
+
+	/**
+	 * (nonpart) Mark the account as participating or non-participating. 
+	 */	@JsonProperty("non-participation")
+	public Boolean nonParticipation;
+
+	/**
+	 * (selkey) Public key used with the Verified Random Function (VRF) result during 
+	 * committee selection. 
+	 */	@JsonProperty("selection-participation-key")
+	public String selectionParticipationKey;
```

Comment:
I wonder what type this should be, digest maybe?

Reply:
Drafted a solution. Should be tested.

Reply:
reverted back to byte[]. This can be done once the full list of Digest cases is identified, and the corresponding e2e test coverage cases are available.